extensions/oni-plugin-quickopen/src/BookmarksSearch.ts

```diff
@@ -0,0 +1,81 @@
+import { Plugin, Menu, Buffer } from "oni-api"
+import { readdir, stat, pathExists } from "fs-extra"
+import { homedir } from "os"
+import { Event, IEvent } from "oni-types"
+import * as path from "path"
+
+import { IAsyncSearch, QuickOpenResult } from "./QuickOpen"
+import { QuickOpenItem, QuickOpenType } from "./QuickOpenItem"
+
+export interface IBookmarkItem {
+    fullPath: string
+    type: string
+}
+
+export default class BookmarkSearch implements IAsyncSearch {
+    private _onSearchResults = new Event<QuickOpenResult>()
+    private _bookmarkItems: QuickOpenItem[]
+
+    constructor(private _oni: Plugin.Api) {
+        const bookmarks = this._oni.configuration.getValue<string[]>("oni.bookmarks", [])
+
+        // TODO: Consider adding folders as well (recursive async with ignores/excludes)
+        this._bookmarkItems = [
+            new QuickOpenItem("Open Folder", "", QuickOpenType.folderHelp),
+            ...bookmarks.map(
+                folder => new QuickOpenItem(folder, "", QuickOpenType.bookmark, folder),
+            ),
+            new QuickOpenItem(
+                "Open configuration",
+                "For adding a bookmark",
+                QuickOpenType.bookmarkHelp,
+            ),
+        ]
+    }
+
+    private _getBookmarkPath = (bookmark: string) => {
+        const userBaseDir = this._oni.configuration.getValue<string>("oni.bookmarks.baseDirectory")
+        const baseDir = userBaseDir || homedir()
```

Comment:
I think if there isn't `"oni.bookmarks.baseDirectory"` defined, then we should default to nothing?

Could cause confusion where I tried setting a bookmark of `F:\User Files\Documents\Git\oni`, which was resolved to `C:\Users\ryan\F:\User Files\Documents\Git\uni_work`.

Or, could leave this for now and in a follow up PR improve the logic here to be:

 * If relative, use base path (either defined, or home dir)
 * If absolute, use that always regardless of the value of `"oni.bookmarks.baseDirectory"`.

I don't mind doing that (along with the globbing), or its also a pretty simple change so we could leave it as a "first issue"?

If you'd rather do that, then I can accept the PR in its current form, since the rest of it works great!

Reply:
@CrossR tweak this do use absolute if absolute and relative resolves from a base dir